ws/types.nim

```diff
@@ -92,10 +92,9 @@ type
 
   Ext* = ref object of RootObj
     name*: string
-    options*: Table[string, string]
 
-  ExtFactory* = proc(name: string, options: Table[string, string]):
-    Ext {.raises: [Defect].}
+  ExtFactory* = proc(serverSide: bool, options: seq[ExtParam]):
```

Comment:
Do we really need to make a distinction for server vs client side?

Reply:
currently I'm using it for optimization and diagnosis purpose, I need more thorough analysis before discard it.